test/test_tensor_creation_ops.py

```diff
@@ -2117,13 +2117,39 @@ def test_constructor_dtypes(self, device):
             self.assertIs(torch.float64, torch.get_default_dtype())
             self.assertIs(torch.cuda.DoubleStorage, torch.Storage)
 
-        # don't support integral or sparse default types.
-        self.assertRaises(TypeError, lambda: torch.set_default_tensor_type('torch.IntTensor'))
-        self.assertRaises(TypeError, lambda: torch.set_default_dtype(torch.int64))
-
         # don't allow passing dtype to set_default_tensor_type
+        self.assertRaises(TypeError, lambda: torch.set_default_tensor_type('torch.IntTensor'))
         self.assertRaises(TypeError, lambda: torch.set_default_tensor_type(torch.float32))
 
+        # don't allow passing dtype to set_default_dtype
+        # see AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_AND_QINTS
+        all_scalar_types_with_complex_and_qints = (
```

Comment:
should we add a method in `torch/testing/_legacy.py` to get the list of all valid dtypes (including quantized types)?

cc. @mruberry

Reply:
fwiw, i'm aware of that file. decided to list them explicitly here as it makes things easier to follow. but maybe it would be better to reuse whatever is in legacy. that said, could this set of types be used anywhere else? i'm not sure. but there are a bunch of places where people define lists like this, so dunno. lmk what you prefer

Reply:
I totally understand. I am suggester to have one getter to have one source of truth. Using these explicitly defined lists got us into this trouble in the firsts place

Reply:
agreed, i'll move these to legacy

Reply:
did the minimal required change. i think this is good enough for now. if others want to use this, they can extend code in the legacy file to match other dtypes